raw_map/src/transform/shrink_roads.rs

```diff
@@ -57,7 +59,13 @@ pub fn shrink(raw: &mut RawMap, timer: &mut Timer) {
         // TODO It'd be better to gradually shrink each road until they stop touching. I got that
         // working in some maps, but it crashes in others (downstream in intersection polygon code)
         // for unknown reasons. Just do the simple thing for now.
-        raw.roads.get_mut(&r1).unwrap().scale_width = 0.5;
-        raw.roads.get_mut(&r2).unwrap().scale_width = 0.5;
+        for id in [r1, r2] {
+            // Anything derived from lane_specs_ltr will need to be changed. When we store
```

Comment:
This was the "big conceptual leap" for this PR. Previously we mutated a `scale_width` on `RawRoad` and haphazardly applied it in some of the query methods. It was confusing to think about _when_ to apply it and make sure not to do it twice. So I actually like this approach more.

Reply:
Looks good. We can fiddle with the individual widths however we like.

We might end up with one set of widths at the start and at the end. See [`placement:{start,end}`](https://taginfo.openstreetmap.org/keys/placement%3Astart) and `width:lanes:{start,end}` which exist as explicit versions of `placement=transition`, and I guess "width = transition", and were crucial parts of JOSM Lanes, and will probably be needed for placement impl here.

The roads can tween between start and end, and we can match the next `end` to the next `start` so that things join up. I wonder we do that to represent the curb shape when a new lane is created...

Reply:
Ooh, I hadn't even thought about that yet. We'll definitely add more detail over time.